internal/services/forwarder/service.go

```diff
@@ -18,51 +20,62 @@ var messageTypeIcons = map[entities.MessageType]string{
 	entities.MessageTypeReply: "↩️",
 }
 
-type Forwarder struct {
-	tgToken  string
-	tgChatId int
-	q        *queue.Queue[entities.Message]
+type Service struct {
+	tgBotToken string
+	q          *queue.Queue[entities.Message]
+	repo       repository.Repository
+	l          *slog.Logger
 }
 
-func New(cfg *Config, queue *queue.Queue[entities.Message]) *Forwarder {
-	return &Forwarder{
-		tgToken:  cfg.TgConfig.Token,
-		tgChatId: cfg.TgConfig.ChatId,
-		q:        queue,
+func New(cfg *Config, queue *queue.Queue[entities.Message], repo repository.Repository, l *slog.Logger) *Service {
+	return &Service{
+		tgBotToken: cfg.BotToken,
+		q:          queue,
+		repo:       repo,
+		l:          l.With("name", "ForwarderService"),
 	}
 }
 
-func (f *Forwarder) Run(ctx context.Context) error {
-	botSettings := tele.Settings{Token: f.tgToken}
+func (s *Service) Run(ctx context.Context) error {
+	botSettings := tele.Settings{Token: s.tgBotToken}
 	bot, err := tele.NewBot(botSettings)
 	if err != nil {
 		return err
 	}
 
-	slog.Info("forwarder is ready", "username", bot.Me.Username)
+	s.l.Info("forwarder is ready", "username", bot.Me.Username)
 
 	for {
 		select {
-		case message := <-f.q.AsChan():
+		case <-ctx.Done():
+			s.l.Info("stopping forwarder service")
+			return nil
+		case message := <-s.q.AsChan():
+			l := s.l.With("interactionId", message.InteractionId, "fromVkSenderId", message.VkSenderId)
+			interaction, err := s.repo.GetInteraction(message.InteractionId)
```

Comment:
(nit) I'm a bit concerned that this is the second repo request while handling the same VK message. First we request the Interaction in the VK webhook handler to check that it exists, and then we request the same Interaction here.

Maybe consider extracting `TgChatId` from the Interaction in the handler and putting it to the queue. I.e. replacing `InteractionId` with `TgChatId` in the `Message` struct

Reply:
No performance issues in foreseeable future, so let it be, as it seems more flexible this way (e.g. may come the need to extract more information in forwarder service)